src/execution_impl.rs

```diff
@@ -83,21 +83,16 @@ pub(crate) fn exec(
     param: &str,
     interface: &dyn Interface,
 ) -> Result<Response> {
+    let mut env = Env::new(interface);
     let instance = match instance {
         Some(instance) => instance,
-        None => create_instance(limit, module, interface)?,
+        None => create_instance(limit, module, env.clone())?,
     };
-    let mut env = as_ffi_bindings::Env::default();
-    env.init(&instance)?;
+    env.init_with_instance(&instance)?;
```

Comment:
The method `init_with_instance` is meant, and is, called internally by wasmer when creating the instance, I don't see a reason to call it here. 

FYI it's called at https://github.com/wasmerio/wasmer/blob/db8c4c5d1ce4c84504e4b2606e0037e72099e374/lib/vm/src/instance/mod.rs#L1258

Reply:
Actually looking at [the ffi docs](https://github.com/cppccn/as-ffi-bindings#env-instantiation), I'd say you need to do a `env.wasm_env.init(&instance)`, since we're using the ffi env directly below with the `StringPtr`.

Reply:
Yes the problem is that we need the env in the StringPtr